[CU-jd835e] Creator view - add scheduling settings

Buglik opened the pull request:
![Capture1](https://user-images.githubusercontent.com/9595999/119571347-88634400-bdb1-11eb-828e-32171221e5d2.PNG)
![Capture2](https://user-images.githubusercontent.com/9595999/119571348-88fbda80-bdb1-11eb-9a5a-9934cb68a7e6.PNG)
![Capture3](https://user-images.githubusercontent.com/9595999/119571350-88fbda80-bdb1-11eb-9586-47c6ddeeccc1.PNG)
![Capture4](https://user-images.githubusercontent.com/9595999/119571351-89947100-bdb1-11eb-910d-fdb988f01b34.PNG)
![Capture5](https://user-images.githubusercontent.com/9595999/119571352-89947100-bdb1-11eb-9505-6f05c2b4a4e2.PNG)
![Capture6](https://user-images.githubusercontent.com/9595999/119571333-87321700-bdb1-11eb-94fe-a32640289fa2.PNG)
![Capture7](https://user-images.githubusercontent.com/9595999/119571340-87caad80-bdb1-11eb-926e-8c81b930a88d.PNG)
![Capture8](https://user-images.githubusercontent.com/9595999/119571342-88634400-bdb1-11eb-9667-d644eaf8c19d.PNG)

Buglik committed:
adding scheduling options to tournamentCreator

Zummek commented:
Task linked: [CU-jd835e Creator view - add scheduling settings](https://app.clickup.com/t/jd835e)

Buglik committed:
Visually ready
changed date format
added locales
changed error messages placement (now at stepper)

Buglik committed:
minor changes after meeting
change days array values
moved tournament name to step1
changed tournament name placement at step3
removed table delete btn in step3

Zummek commented on web-app/src/components/tournament/creator/DaysOfPlaySelector.vue:
```diff
@@ -0,0 +1,73 @@
+
+@Component
+export default class DaysOfPlaySelector extends Vue {
+  @Prop({ type: Array, required: true })
+  days!: [];
+
+  private pickedDays = this.days;
```
If you want, u can make that through SyncProp or PropSync. You can find an example of that in our project.
As a result, I think we can get less code

Zummek replied:
https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier

Zummek commented on web-app/src/components/tournament/creator/FrequencySelector.vue:
```diff
@@ -0,0 +1,32 @@
+
+  updateDays() {
+    this.$emit('update-freq', this.newFreq);
+  }
```
Same suggestion as above(SyncProp)

Zummek commented on web-app/src/components/tournament/creator/StartDateSelector.vue:
```diff
@@ -0,0 +1,47 @@
+
+  updateDate() {
+    this.$emit('update-date', this.newDate);
+  }
```
same

Zummek commented on web-app/src/components/tournament/creator/TournamentNameChanger.vue:
```diff
@@ -0,0 +1,82 @@
+
+  updateTournamentName() {
+    this.$emit('name-changed', this.newTournName);
+  }
```
same

Zummek commented on web-app/src/components/tournament/creator/TournamentTypeSelectorItem.vue:
```diff
@@ -1,14 +1,14 @@
   <div class="row col-8 col-sm-4 justify-center q-ma-md">
-    <q-btn class="row col-12" no-caps padding="none">
+    <q-btn class="row col-12 selectorButton" no-caps padding="none">
       <q-card
         flat
-        class="col-12 tournamentTypeCard"
+        class="row col-12 tournamentTypeCard items-center"
         :class="tournamentType === activeType ? 'activeTypeCard' : ''"
         @click="updateTournamentType"
       >
```
When did you set height/width through `selectorButton` to `q-btn` is required to set this same to child component `q-card`?

Buglik replied:
When not set, q-card doesn't take full btn space, pic below 
![obraz](https://user-images.githubusercontent.com/9595999/120507313-fc5fa680-c3c6-11eb-8b38-7e41d8c2798d.png)

Zummek commented on web-app/src/pages/tournament/TournamentCreator.vue:
```diff
@@ -225,6 +253,10 @@ export default class TournamentCreator extends Vue {
 
   private async submitAddTournament() {
+    const dateToSubmit = new Date(Date.parse(this.startDateString));
+    // console.log(dateToSubmit);
+    // console.log(this.days);
+    // console.log(this.frequency)
```
.

Buglik replied:
leaving comments for now, these are not connected to api yet

Buglik committed:
changed props to propsSync
